### ISSUE TITLE
Tag search service not siteaccess aware

### ISSUE THREAD

**Issue by Mrkisha** [netgen/TagsBundle#56]
This might be related to previous issue #55 .

Here is the scenario:
```cucumber
    Background:
        Given there is "siteaccess1" and "siteaccess2"
        And they both use language "language1"
        And there is an Article content type with tag field
        And tags "tag1", "tag2" and "tag3" exist for "language1"
    
    Scenario: 
        Given I create "article1" in "siteaccess1"
        And I add "tag1" to this "article1"
        And I create "article2" in "siteaccess2"
        And I add "tag1", "tag2" and "tag3" to "article2"
        When I use netgen's api service tag to search for related content 
            in "siteaccess1" for "tag1"
        Then I should see only "article1" in returned results
```
Basically this is the `PHP` code I'm using:
```php
    $tagService = $this->get('eztags.api.service.tags');
    /** @var \Netgen\TagsBundle\API\Repository\Values\Tags\SearchResult $results */
    $results = $tagService->searchTags($tag, $language);
```

Search **is** language aware but **not** siteaccess aware.

**Comment by Mrkisha** [comment]
The workaround I have just implemented goes like this:

```php
$repo = //...

$searchResult = $repo->getSearchService()->findContent(
    new Query(
        array(
            'filter' => new TagId($tagObject->id),
            'sortClauses' => array(
                new Query\SortClause\DateModified(Query::SORT_DESC),
            ),
            'query' => new Criterion\LogicalAnd([
                new Criterion\Visibility(Criterion\Visibility::VISIBLE),
                new Criterion\LanguageCode($lang),
                new Criterion\ContentTypeIdentifier(['some_content_type']),
                new Criterion\Subtree($this->getRootLocation()->pathString)
            ]),
        )
    )
);
```

this way I get content that is just from current `siteaccess` and the respective language.

**Comment by emodric** [comment]
This will have to wait until there's an official eZ site API as to not reinvent the wheel, but rather use/mimic official APIs.

**Comment by andrerom** [comment]
@emodric : https://github.com/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/Core/Repository/SiteAccessAware  _(naming wise it's only SALanguageAware atm fyi, but that seems to be what is asked for here)_

**Comment by emodric** [comment]
Is it? @Mrkisha in his second example uses `$repo->getSearchService()->findContent()`, so I'm not sure what was asked, to be honest :)

**Comment by andrerom** [comment]
True, he adds subtree filtering to the mix which will probably newer be the case out of the box on findContent(), but rather higher level SiteService for instance that does these things for you.

**Comment by emodric** [comment]
Fixed by #103
